pkg/webhook/utils.go

```diff
@@ -51,6 +54,7 @@ func CreateVolumeForSecret(secret *core.Secret) corev1.Volume {
 						Path: strings.ToLower(secret.Key),
 					},
 				},
+				Optional: &optional,
```

Comment:
While we are at it, can we make it read only as well?

Reply:
So when we actually [mount the volume](https://github.com/flyteorg/flytepropeller/blob/560bb1b0c198c2e7074e2f6aa74d45d933fbe846/pkg/webhook/utils.go#L62) it is already designated as ReadOnly. Does this cover your comment? It looks like on the `SecretVolumeSource ` there is a `DefaultMode *int32` field where we can use octal value (ie. 0555) to designate default mount permissions, but I'm not sure it's necessary.

Reply:
Ah, we are good then!